client/webserver/site/src/js/forms.ts

```diff
@@ -153,15 +153,21 @@ export class NewWalletForm {
     const { asset, parentAsset } = this.current
     const selectedDef = this.current.selectedDef
     let parentForm
+    let walletType = selectedDef.type
     if (parentAsset) {
+      if (!asset.token) {
+        this.setError('token is absent from asset with parent, should never happen!')
+        return
+      }
```

Comment:
Like you said, it should never happen. We don't need defensive code for this. If you're just trying to deal with TypeScript complaining about the optional field, you can do `if (parentAsset && asset.token)`

Reply:
Maybe just do `walletType = (asset.token as Token).definition.type` then?

Reply:
> Maybe just do `walletType = (asset.token as Token).definition.type` then?

That's probably better.